### PR TITLE
Create app with prettier eslint

### PR DESCRIPTION
**Pull request by zigasinko**
Created the application with create-react-app (typescript). Added ESlint and prettier with initial configurations.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,26 @@
...
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
```
Comment (dblenkus) [inline .gitignore]: Add a new line here.

File: public/index.html [hunk 2]
```diff
@@ -21,12 +21,12 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-        <title>React App</title>
-    </head>
-    <body>
-        <noscript>You need to enable JavaScript to run this app.</noscript>
-        <div id="root"></div>
-        <!--
+    <title>dictyExpress</title>
```
Comment (dblenkus) [inline public/index.html]: This is not related to ESlint.

File: package.json [hunk 3]
```diff
@@ -1,39 +1,43 @@
...
-            "not dead",
-            "not op_mini all"
-        ],
-        "development": [
-            "last 1 chrome version",
-            "last 1 firefox version",
-            "last 1 safari version"
-        ]
-    }
+  "name": "dicty-express",
```
Comment (dblenkus) [inline package.json]: Why is this not a part of the previous commit?

**Review by dblenkus — COMMENTED** [review]
https://chris.beams.io/posts/git-commit/#end

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+module.exports = {
+    env: {
+        "browser": true,
+        "jest": true
+    },
+  parser: "@typescript-eslint/parser", // Specifies the ESLint parser
+  extends: [
+    "airbnb",
+    "airbnb/hooks",
+    "plugin:react/recommended", // Uses the recommended rules from @eslint-plugin-react
```
Comment (dblenkus) [inline .eslintrc.js]: This is already included in the airbnb configuration: https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb#eslint-config-airbnb-1

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    "prettier/react",
+    "plugin:@typescript-eslint/recommended", // Uses the recommended rules from the @typescript-eslint/eslint-plugin
```
Comment (dblenkus) [inline .eslintrc.js]: I would prefer to have this one at the top to avoid overriding too much of the airbnb configuration.

File: .eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    ecmaFeatures: {
+      jsx: true, // Allows for the parsing of JSX
+    },
+  },
+  rules: {
+    "react/jsx-filename-extension": [1, { extensions: [".tsx"]}],  // .tsx files can include JSX
+    "import/extensions": ["error", "never", { "svg": "always" }]  // Ensure consistent use of file extension within import path
+  },
+  settings: {
+    react: {
```
Comment (dblenkus) [inline .eslintrc.js]: This is already in the airbnb vonfiguration: https://github.com/airbnb/javascript/blob/06b3ab11d9a443ff46f052dd00375e271e5146e6/packages/eslint-config-airbnb/rules/react.js#L542

File: package.json [hunk 7]
```diff
@@ -1,39 +1,43 @@
...
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "^4.2.4",
+    "@testing-library/react": "^9.5.0",
+    "@testing-library/user-event": "^7.2.1",
+    "@types/jest": "^24.9.1",
+    "@types/react": "^16.9.32",
+    "@types/react-dom": "^16.9.6",
+    "eslint-config-airbnb": "^18.1.0",
```
Comment (dblenkus) [inline package.json]: All linter and auto-formatting dependencies should go under dev dependencies.

Comment (dblenkus) [reply package.json]: You should add `eslint` requirement, as you reference it in our files, i.e. in `package.json`.

File: .eslintrc.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+module.exports = {
+    env: {
+        "browser": true,
+        "jest": true
+    },
+  parser: "@typescript-eslint/parser", // Specifies the ESLint parser
```
Comment (dblenkus) [inline .eslintrc.js]: Comments should go into a separate line above the line that they refer to. Also, add a dot at the end.

File: .eslintrc.js (hunk 5 above)
Comment (dblenkus) [reply .eslintrc.js]: Also add `plugin:@typescript-eslint/recommended-requiring-type-checking` as our codebase will be relatively small.

File: .eslintrc.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  ],
+  parserOptions: {
+    ecmaVersion: 2018, // Allows for the parsing of modern ECMAScript features
+    sourceType: "module", // Allows for the use of imports
+    ecmaFeatures: {
+      jsx: true, // Allows for the parsing of JSX
+    },
+  },
+  rules: {
+    "react/jsx-filename-extension": [1, { extensions: [".tsx"]}],  // .tsx files can include JSX
```
Comment (dblenkus) [inline .eslintrc.js]: `1` -> `warn`

**Comment by dblenkus** [comment]
I also suggest adding the following `.prettierrc.js` configuration:
```
module.exports = {
    semi: true,
    trailingComma: 'all',
    singleQuote: true,
    printWidth: 100,
    tabWidth: 4,
};
```

File: public/index.html (hunk 2 above)
Comment (zigasinko) [reply public/index.html]: Yes, just a minor beauty fix, will put this type of fixed in a separate commit in the future.

File: package.json (hunk 3 above)
Comment (zigasinko) [reply package.json]: It slipped through the cracks. Just a minor fix, not really relevant.

File: .eslintrc.js (hunk 4 above)
Comment (zigasinko) [reply .eslintrc.js]: Removed everything from react plugin.

File: .eslintrc.js (hunk 6 above)
Comment (zigasinko) [reply .eslintrc.js]: Removed.

File: .eslintrc.js (hunk 8 above)
Comment (zigasinko) [reply .eslintrc.js]: Done.

File: package.json (hunk 7 above)
Comment (zigasinko) [reply package.json]: Besides all mentioned above, I've also moved all testing libraries. Installed eslint.

File: .eslintrc.js (hunk 5 above)
Comment (zigasinko) [reply .eslintrc.js]: Added `plugin:@typescript-eslint/recommended-requiring-type-checking`.
Every how-to, even the airbnb one (https://github.com/iamturns/eslint-config-airbnb-typescript) suggest adding typescript-eslint/recommended after airbnb one.

File: .eslintrc.js (hunk 9 above)
Comment (zigasinko) [reply .eslintrc.js]: Changed to "error".

**Comment by zigasinko** [comment]
Force pushed code-review optimizations!

File: .eslintrc.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  env: {
+      "browser": true,
+      "jest": true
+  },
+  // Specifies the ESLint parser. Allows ESLint to understand TypeScript syntax.
+  parser: "@typescript-eslint/parser",
+  // Allows us to use rules within our codebase.
+  plugins: ["@typescript-eslint", "prettier"],
+  extends: [
+    // Config rules specified mostly from airbnb-config-typescript GitHub: https://github.com/iamturns/eslint-config-airbnb-typescript and https://blog.geographer.fr/eslint-guide.
```
Comment (dblenkus) [inline .eslintrc.js]: Keep comments under 80 characters. You can split them into multiple lines.

File: public/index.html [hunk 11]
```diff
@@ -1,18 +1,18 @@
...
 <html lang="en">
-    <head>
-        <meta charset="utf-8" />
-        <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
-        <meta name="viewport" content="width=device-width, initial-scale=1" />
-        <meta name="theme-color" content="#000000" />
-        <meta name="description" content="Web site created using create-react-app" />
-        <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
-        <!--
+  <head>
```
Comment (dblenkus) [inline public/index.html]: Do we have 2- or 4-spaces indentation in HTML files? I would prefer 4 to keep it in sync with tsx files as they also have HTML-like syntax.

File: public/index.html (hunk 2 above)
Comment (dblenkus) [reply public/index.html]: You can move it into the previous commit or create a new one.

**Review by dblenkus — APPROVED** [review]
Looks good, I only have few minor comments.

File: public/index.html (hunk 11 above)
Comment (zigasinko) [reply public/index.html]: Added npm scripts to check/fix all files, not just .js, .jsx, .ts and .tsx (which are handled by ESLint).
